### DISCUSSION TITLE
How to define a searchable select filter?

### DISCUSSION THREAD

**Discussion by piljac1** [laravel/nova-issues#5086] [Q&A]
According to issue https://github.com/laravel/nova-issues/issues/1894, it was implemented in Nova 4, but it doesn't seem to be documented. 

Any help is appreciated!

Thanks!

**Comment by crynobone** [comment]
It's exclusively for Filterable Field, e.g:

```php
Select::make('Types')->filterable()->searchable(),
```

**Reply by piljac1** [reply]
Alright. Can we hope for a custom filter (filter class) searchable implementation anytime soon, or should we go custom ? First thing I've tried was to add a `field` array with `'searchable' => true` in `jsonSerialize` (because it looked like that's what `isSearchable` was looking for on the Vue side of things), but it didn't seem to work. Any insight on how to achieve the searchable behavior with minimum customization ? Thanks in advance :)

**Reply by crynobone** [reply]
No plan for that. You can always create custom filter to suit your implementation.

**Reply by piljac1** [reply]
I will try that tomorrow thanks. On another note, I'm here working on a custom filter because I tried to use `->filterable()->searchable()` on a `HasOneThrough` field, which is not supported. In that regard, is `HasOneThrough` filterable support is planned on your end?

**Reply by crynobone** [reply]
No plan as well.

**Comment by ninthspace** [comment]
I've worked around this by having a `Text` field and another field that `dependsOn` on it (e.g. a `MultiSelect` or `BooleanGroup`), so when the `Text` field changes, the options inside the other field are recomputed and displayed. You can hide or show the dependent field according to whether there are any matches.
